packages/authorization-process/src/utilities/errorMappers.ts

```diff
@@ -64,5 +65,25 @@ export const addUserErrorMapper = (error: ApiError<ErrorCodes>): number =>
     .with("clientNotFound", () => HTTP_STATUS_NOT_FOUND)
     .with("securityUserNotFound", () => HTTP_STATUS_NOT_FOUND)
     .with("userAlreadyAssigned", () => HTTP_STATUS_FORBIDDEN)
+    .otherwise(() => HTTP_STATUS_INTERNAL_SERVER_ERROR);
 
+export const addClientPurposeErrorMapper = (
+  error: ApiError<ErrorCodes>
+): number =>
+  match(error.code)
+    .with(
+      "clientNotFound",
+      "purposeNotFound",
+      "eserviceNotFound",
+      "agreementNotFound",
+      "descriptorNotFound",
+      "noVersionsFoundInPurpose",
```

Comment:
`agreementNotFound` and `noVersionsFoundInPurpose` should be `400`, the others `500`

Reply:
correct.
Just to avoid misunderstanding, the comment starts at row 77, so `clientNotFound` and `purposeNotFound` will still be 404